--- a/include/configs/tqmls102xa.h	2017-03-13 12:37:51.066020614 +0100
+++ b/include/configs/tqmls102xa.h	2017-03-14 09:35:15.942062154 +0100
@@ -52,7 +52,6 @@
 #endif
 
 /* XHCI Support - enabled by default */
-#define CONFIG_HAS_FSL_XHCI_USB
 
 #ifdef CONFIG_HAS_FSL_XHCI_USB
 #define CONFIG_USB_XHCI_FSL
@@ -295,7 +294,6 @@
 #endif /* eTSEC */
 
 /* PCIe */
-#define CONFIG_PCI		/* Enable PCI/PCIE */
 #define CONFIG_PCIE1		/* PCIE controler 1 */
 #define CONFIG_PCIE2		/* PCIE controler 2 */
 #define CONFIG_PCIE_LAYERSCAPE	/* Use common FSL Layerscape PCIe code */
@@ -361,7 +359,7 @@
 #define CONFIG_HWCONFIG
 #define HWCONFIG_BUFFER_SIZE		128
 
-#define CONFIG_BOOTDELAY		3
+#define CONFIG_BOOTDELAY		1
 
 #ifdef CONFIG_LPUART
 #define TQMLS201X_CONSOLE_DEV		"ttyLP0"
@@ -384,7 +382,6 @@
 
 #define CONFIG_CMD_GREPENV
 #define CONFIG_CMD_MEMINFO
-#define CONFIG_CMD_MEMTEST
 #define CONFIG_SYS_MEMTEST_START	0x80000000
 #define CONFIG_SYS_MEMTEST_END		0x9fffffff
 
@@ -478,122 +475,21 @@
 	"kernel_start="__stringify(TQMLS102X_KERNEL_SECTOR_START)"\0"          \
 	"kernel_size="__stringify(TQMLS102X_KERNEL_SECTOR_COUNT)"\0"           \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0"                       \
-	"loadimage=mmc dev ${mmcdev}; "                                        \
-		"mmc read ${loadaddr} ${kernel_start} ${kernel_size};\0"       \
-	"loadfdt=mmc dev ${mmcdev}; "                                          \
-		"mmc read ${fdt_addr} ${fdt_start} ${fdt_size};\0"             \
-	"update_uboot=if tftp ${uboot}; then "                                 \
-		"if itest ${filesize} > 0; then "                              \
-			"mmc dev ${mmcdev}; mmc rescan; "                      \
-			"setexpr blkc ${filesize} / 0x200; "                   \
-			"setexpr blkc ${blkc} + 1; "                           \
-			"if itest ${blkc} <= ${uboot_size}; then "             \
-				"mmc write ${loadaddr} ${uboot_start} "        \
-					"${blkc}; "                            \
-			"fi; "                                                 \
-		"fi; fi; "                                                     \
-		"setenv filesize; setenv blkc \0"                              \
-	"update_kernel=run kernel_name; "                                      \
-		"if tftp ${kernel}; then "                                     \
-			"if itest ${filesize} > 0; then "                      \
-				"mmc dev ${mmcdev}; mmc rescan; "              \
-				"setexpr blkc ${filesize} / 0x200; "           \
-				"setexpr blkc ${blkc} + 1; "                   \
-				"if itest ${blkc} <= ${kernel_size}; then "    \
-					"mmc write ${loadaddr} "               \
-						"${kernel_start} ${blkc}; "    \
-				"fi; "                                         \
-			"fi; "                                                 \
-		"fi; "                                                         \
-		"setenv filesize; setenv blkc \0"                              \
-	"update_fdt=if tftp ${fdt_file}; then "                                \
-		"if itest ${filesize} > 0; then "                              \
-			"mmc dev ${mmcdev}; mmc rescan; "                      \
-			"setexpr blkc ${filesize} / 0x200; "                   \
-			"setexpr blkc ${blkc} + 1; "                           \
-			"if itest ${blkc} <= ${fdt_size}; then "               \
-				"mmc write ${loadaddr} ${fdt_start} ${blkc}; " \
-			"fi; "                                                 \
-		"fi; fi; "                                                     \
-		"setenv filesize; setenv blkc \0"                              \
 	BASEBOARD_EXTRA_ENV_SETTINGS                                           \
 
 #define CONFIG_BOOTCOMMAND \
-	"run mmcboot; run netboot; run panicboot"
+	"run defaultboot"
 
 #define CONFIG_EXTRA_ENV_SETTINGS                                              \
-	"addmisc=setenv bootargs ${bootargs} hdmi\0"                           \
 	"board=tqmls102x\0"                                                    \
-	"uimage=uImage\0"                                                      \
-	"zimage=zImage\0"                                                      \
 	"boot_type=bootz\0"                                                    \
-	"kernel_name=if test \"${boot_type}\" != bootz; then "                 \
-		"setenv kernel ${uimage}; "                                    \
-		"else setenv kernel ${zimage}; fi\0"                           \
-	"uboot=" CONFIG_UBOOT_MMCSD_IMAGE_FILE "\0"                            \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0"                               \
 	"fdt_addr="__stringify(TQMLS102X_FDT_ADDRESS)"\0"                      \
 	"console=" TQMLS201X_CONSOLE_DEV "\0"                                  \
 	"fdt_high=0xffffffff\0"                                                \
 	"initrd_high=0xffffffff\0"                                             \
-	"addtty=setenv bootargs ${bootargs} console=tty0 "                     \
-		"console=${console},${baudrate}\0"                             \
-	"mmcpart=2\0"                                                          \
-	"mmcblkdev=0\0"                                                        \
-	"mmcargs=run addmmc addtty addmisc addplatform\0"                      \
-	"addmmc=setenv bootargs ${bootargs} "                                  \
-		"root=/dev/mmcblk${mmcblkdev}p${mmcpart} rw rootwait\0"        \
-	"mmcboot=echo Booting from mmc ...; "                                  \
-		"setenv bootargs; "                                            \
-		"run mmcargs; "                                                \
-		"run loadimage; "                                              \
-		"if run loadfdt; then "                                        \
-			"echo boot device tree kernel ...; "                   \
-			"${boot_type} ${loadaddr} - ${fdt_addr}; "             \
-		"else "                                                        \
-			"${boot_type}; "                                       \
-		"fi;\0"                                                        \
-		"setenv bootargs \0"                                           \
-	"netdev=eth0\0"                                                        \
-	"rootpath=/srv/nfs/tqmls1021a\0"                                       \
-	"ipmode=static\0"                                                      \
-	"netargs=run addnfs addip addtty addmisc\0"                            \
-	"addnfs=setenv bootargs ${bootargs} "                                  \
-		"root=/dev/nfs rw "                                            \
-		"nfsroot=${serverip}:${rootpath},v3,tcp;\0"                    \
-	"addip_static=setenv bootargs ${bootargs} "                            \
-		"ip=${ipaddr}:${serverip}:${gatewayip}:${netmask}:"            \
-		"${hostname}:${netdev}:off\0"                                  \
-	"addip_dynamic=setenv bootargs ${bootargs} ip=dhcp\0"                  \
-	"addip=if test \"${ipmode}\" != static; then "                         \
-		"run addip_dynamic; else run addip_static; fi\0"               \
-	"set_getcmd=if test \"${ipmode}\" != static; then "                    \
-		"setenv getcmd dhcp; setenv autoload yes; "                    \
-		"else setenv getcmd tftp; setenv autoload no; fi\0"            \
-	"netboot=echo Booting from net ...; "                                  \
-		"run kernel_name; "                                            \
-		"run set_getcmd; "                                             \
-		"setenv bootargs; "                                            \
-		"run netargs; "                                                \
-		"if ${getcmd} ${kernel}; then "                                \
-			"if ${getcmd} ${fdt_addr} ${fdt_file}; then "          \
-				"${boot_type} ${loadaddr} - ${fdt_addr}; "     \
-			"fi; "                                                 \
-		"fi; "                                                         \
-		"echo ... failed\0"                                            \
-	"panicboot=echo No boot device !!! reset\0"                            \
-	"uboot-qspi=" CONFIG_UBOOT_QSPI_IMAGE_FILE "\0"                        \
-	"update_uboot-qspi=if tftp ${uboot-qspi}; then "                       \
-		"if itest ${filesize} > 0; then "                              \
-			"sf probe; sf update ${loadaddr} "__stringify(TQMLS102X_UBOOT_SPI_OFFSET)" ${filesize}; " \
-		"fi; fi; "                                                     \
-		"setenv filesize;\0"                                           \
-	"rcw-qspi=ls102xa-rcw-tqmls-qspi-0100.bin.bswap\0"                     \
-	"update_rcw=if tftp ${rcw-qspi}; then "                                \
-		"if itest ${filesize} > 0; then "                              \
-			"sf probe; sf update ${loadaddr} 0 ${filesize}; "      \
-		"fi; fi; "                                                     \
-		"setenv filesize;\0"                                           \
+	"ethact=eTSEC2\0"                                                      \
+	"defaultboot=sf probe; sf read 0x82000000 0x150000 0x500000; sf read 0x88000000 0x140000 0x10000; setenv bootargs console=ttyS0,115200 rootfstype=squashfs root=/dev/mtdblock5 quiet systemd.show_status=auto libata.force=disable; bootz 0x82000000 - 0x88000000\0" \
 	TQMLS102X_EXTRA_BOOTDEV_ENV_SETTINGS                                   \
 
 #define CONFIG_CMD_BOOTZ
