From 562435f1a719a06fd98f4f9135f00fc2645d6a18 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
Date: Fri, 22 Jan 2016 14:04:46 +0100
Subject: [PATCH] mtd: fsl-qspi: accept single and multi io RDSR commands

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index d1109a9..81122d9 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -526,6 +526,7 @@ static int fsl_qspi_get_seqid(struct fsl_qspi *q, u8 cmd)
 	case SPINOR_OP_PP:
 		return SEQID_PP;
 	case SPINOR_OP_RDID:
+	case SPINOR_OP_MIO_RDID:
 		return SEQID_RDID;
 	case SPINOR_OP_WRSR:
 		return SEQID_WRSR;
