From c9c512554a3d08988266a813c72753350c658938 Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
Date: Thu, 28 Apr 2016 18:32:40 +0200
Subject: [PATCH] Makefile: release TQMLS102xA-BSP-0100

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 70dea02..b3bea77 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION = -TQMLS102xA-BSP-0100
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
